### PR TITLE
Fix memory leak by selecting files

### PR DESCRIPTION
**Pull request by Danue1**
before
after

I fixed an issue where the input element was continuously added every time I tried to select an image.

### CODE REVIEW

**Review by talyguryn — APPROVED** [review]

**Comment by Danue1** [comment]
Ah, wait!
inputElement is not assigned.

**Comment by Danue1** [comment]
Ok, got ready.

File: src/utils.js [hunk 1]
```diff
@@ -119,55 +119,76 @@ module.exports = class Utils {
...
+        /**
+         * Set a 'FILE' type for this input element
+         * @type {string}
+         */
+        inputElement.type = 'file';
 
-      if (config.accept) {
-        inputElement.setAttribute('accept', config.accept);
-      }
+        if (config.multiple) {
```
Comment (neSpecc) [inline src/utils.js:137]: config will be accepted only for the first calling?

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/utils.js [hunk 2]
```diff
@@ -119,55 +119,76 @@ module.exports = class Utils {
...
+       * Check
        */
-      let inputElement = document.createElement('INPUT');
-
-      /**
-       * Set a 'FILE' type for this input element
-       * @type {string}
-       */
-      inputElement.type = 'file';
+      if (input === null) {
```
Comment (talyguryn) [inline src/utils.js:124]: image

**Review by talyguryn — CHANGES_REQUESTED** [review]
Example doesn't work. Run `yarn start`, open [localhost:3000](http://localhost:3000) and try to upload image.

File: src/utils.js (hunk 2 above)
Comment (Danue1) [reply src/utils.js]: Oops, sorry.
I commited aboute this.

File: src/utils.js (hunk 1 above)
Comment (Danue1) [reply src/utils.js]: That's right.
I didn't consider that part.
I modified and committed.

**Comment by Danue1** [comment]
Is there anything else I missed?

**Comment by talyguryn** [comment]
Run `yarn start`, open `http://localhost:3000` and try your fixed library on the example page.

image

**Comment by Danue1** [comment]
> Run `yarn start`, open `http://localhost:3000` and try your fixed library on the example page.
> 
> image

@talyguryn 
Oh, I saw it and fixed it.
Now, it seems to be working all correctly if I try to upload image.

File: src/utils.js [hunk 3]
```diff
@@ -119,55 +119,108 @@ module.exports = class Utils {
   static selectFiles(config = {}) {
     return new Promise((resolve, reject) => {
       /**
-       * Create a new INPUT element
-       * @type {HTMLElement}
+       * Check
```
Comment (talyguryn) [inline src/utils.js:122]: the last one. check what?

Comment (Danue1) [reply src/utils.js]: Sorry, I'm late.
I edited the comments, would it be okay like this?

**Review by talyguryn — APPROVED** [review]
Nice

**Comment by Danue1** [comment]
Yeah, Good.

**Comment by Danue1** [comment]
@neSpecc 
Can we merge this?

File: src/utils.js [hunk 4]
```diff
@@ -119,55 +119,109 @@ module.exports = class Utils {
...
+           */
+          inputElement.removeEventListener('change', resolveFiles);
+
+          /**
+           * Remove element from a DOM
+           */
+          document.body.removeChild(inputElement);
+
+          /**
+           * Remove input element
```
Comment (neSpecc) [inline src/utils.js:175]: but it is not removing of the element

Comment (Danue1) [reply src/utils.js:175]: Hmm, Could you recommend a better comment?
I think... "Clear input element for preventing memory leak" is better.

**Pull request closed without merging**